server/service/client.go

```diff
@@ -832,3 +853,179 @@ func extractTmSpecsMacOSSetup(tmSpecs []json.RawMessage) map[string]*fleet.MacOS
 	}
 	return m
 }
+
+// DoGitOps applies the GitOps config to Fleet.
+func (c *Client) DoGitOps(
+	ctx context.Context,
+	config *spec.GitOps,
+	baseDir string,
+	logf func(format string, args ...interface{}),
+	dryRun bool,
+) error {
+	var err error
+	logFn := func(format string, args ...interface{}) {
+		if logf != nil {
+			logf(format, args...)
+		}
+	}
+	group := spec.Group{}
+	scripts := make([]interface{}, len(config.Controls.Scripts))
+	for i, script := range config.Controls.Scripts {
+		scripts[i] = *script.Path
+	}
+	var mdmAppConfig map[string]interface{}
+	var team map[string]interface{}
+	if config.TeamName == nil {
+		group.AppConfig = config.OrgSettings
+		group.EnrollSecret = &fleet.EnrollSecretSpec{Secrets: config.OrgSettings["secrets"].([]*fleet.EnrollSecret)}
+		group.AppConfig.(map[string]interface{})["agent_options"] = config.AgentOptions
+		delete(config.OrgSettings, "secrets")
```

Comment:
Curious why `delete`?

Reply:
Secrets are applied separately with `Client.ApplyGroup`, and it will trigger a validation fail if I don't delete them here.